linkerd/proxy/transport/src/lib.rs

```diff
@@ -85,3 +86,14 @@ fn set_user_timeout_or_warn(
     let stream: std::net::TcpStream = socket2::Socket::into(sock);
     tokio::net::TcpStream::from_std(stream)
 }
+
+#[cfg(not(target_os = "linux"))]
+fn set_user_timeout_or_warn(
+    tcp: TcpStream,
+    user_timeout: Option<Duration>,
+) -> io::Result<TcpStream> {
+    if user_timeout.is_some() {
+        tracing::warn!("TCP_USER_TIMEOUT is supported on Linux only.");
```

Comment:
Eh, won't this print a warning on every single connection? Undesirable.
```suggestion
        tracing::debug!("TCP_USER_TIMEOUT is supported on Linux only.");
```